code/modules/xenoarchaeology/xenoartifact.dm

```diff
@@ -0,0 +1,489 @@
+    
+/obj/item/xenoartifact/proc/get_proximity(range) //Gets a singular bam beano
+    for(var/mob/living/M in range(range, get_turf(src)))
+        return process_target(M)
+    if(isliving(loc))
+        return loc
+
+/obj/item/xenoartifact/proc/get_trait(typepath) //Returns the desired trait and it's values if it's in the artifact's
+    for(var/datum/xenoartifact_trait/T in traits)
+        if(istype(T, typepath))
+            return T
+    return FALSE
+
+/obj/item/xenoartifact/proc/generate_icon(var/icn, var/icnst = "", colour) //Add extra icon overlays
+    icon_overlay = mutable_appearance(icn, icnst)
+    icon_overlay.layer = FLOAT_LAYER
+    icon_overlay.appearance_flags = RESET_ALPHA// Not doing this fucks the alpha?
+    icon_overlay.alpha = alpha//
+    if(colour)
+        icon_overlay.color = colour
+    src.add_overlay(icon_overlay)
+
+/obj/item/xenoartifact/proc/process_target(atom/target) //Hand holding is the best defence
+    if(!istype(target, /mob/living))
+        return target
+    var/mob/living/victim = target
+    if(victim.pulling && istype(victim.pulling, /mob/living))
+        return victim.pulling
+    return victim
+
+/obj/item/xenoartifact/proc/create_beam(atom/target) //Helps show how the artifact is working. Hint stuff.
+    var/datum/beam/xenoa_beam/B = new(src.loc, target, time=1.5 SECONDS, beam_icon='icons/obj/xenoarchaeology/xenoartifact.dmi', beam_icon_state="xenoa_beam", btype=/obj/effect/ebeam/xenoa_ebeam, col = material)
+    INVOKE_ASYNC(B, /datum/beam/xenoa_beam.proc/Start)
+
+/obj/item/xenoartifact/proc/default_activate(chr, mob/user) //used for some stranger cases. Item specific cases that don't fall under the default templates. See battery activator.
+    if(!(manage_cooldown(TRUE)))
+        return
+    charge = chr
+    if(user)
+        true_target += list(process_target(user))
+        check_charge(user)
+        return
+    true_target += list(get_proximity(max_range))
+    check_charge()
+
+/obj/item/xenoartifact/proc/set_frequency(new_frequency)
+	SSradio.remove_object(src, frequency)
+	frequency = new_frequency
+	radio_connection = SSradio.add_object(src, frequency, RADIO_SIGNALER)
+
+/obj/item/xenoartifact/proc/send_signal(var/datum/signal/signal)
+    if(!radio_connection||!signal)
+        return
+    radio_connection.post_signal(src, signal)
+
+/obj/item/xenoartifact/receive_signal(datum/signal/signal)
+    if(!signal || signal.data["code"] != code)
+        return
+    var/mob/living/M = isliving(signal.source.loc) ? signal.source.loc : null
+    audible_message("[icon2html(src, hearers(src))] *beep* *beep* *beep*", null, 3)
+    playsound(get_turf(src), 'sound/machines/triple_beep.ogg', ASSEMBLY_BEEP_VOLUME, TRUE)
+    for(var/datum/xenoartifact_trait/T in traits)
+        if(charge += EASY*T.on_signal(src))
+            true_target += list(get_proximity(max_range))
+            check_charge(M)
+
+/obj/item/xenoartifact/on_block(mob/living/carbon/human/owner, atom/movable/hitby)
```

Comment:
Can this even block? It has no block power

Reply:
Not on its own, no. There are certain traits that allow it to, kinda, block. I'd like to balance them a little more, their values.